types/src/v2/error.rs

```diff
@@ -1,14 +1,24 @@
+use crate::JsonValue;
+use serde::{de::Deserializer, ser::Serializer, Deserialize};
+use std::fmt;
 use thiserror::Error;
 
-/// Error.
-#[derive(Error, Debug)]
-pub enum RpcError {
-	/// Unknown error.
-	#[error("unknown rpc error")]
-	Unknown,
-	/// Invalid params in the RPC call.
-	#[error("invalid params")]
-	InvalidParams,
+/// [JSON-RPC Error object](https://www.jsonrpc.org/specification#error_object)
+#[derive(Error, Debug, PartialEq, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct JsonRpcErrorObject {
+	/// Error code
+	pub code: ErrorCode,
+	/// Message
+	pub message: String,
```

Comment:
This message is totally useless, the message is deduced from the error code AFAIU

However, it needs to be sent sent/received according the spec. Maybe include in ErrorCode deser/ser impls.